[Lens] Add more Lens QA doc (#105057)
elastic/kibana: docs/user/dashboard/lens.asciidoc

[[why-my-field-x-is-missing-from-the-fields-list]]

> ===== Why is my field X missing from the fields list?

Reviewer: Isn't this just `Why is my field missing from the list of fields?`

When I read `field X`, I think `x-axis`.

Author: I see `X` as generic term, but I see what you mean. Will remove it.

> If your field is not in the `Available fields` list in *Lens* there may be multiple reasons, here described:

Reviewer: ```suggestion
Fields do not appear in the *Available fields* in the following scenarios:
```

> * the field is a full-text field, therefore it is not showing in the data summary.

Reviewer: ```suggestion
* The field is a full-text field, therefore it is not showing in the data summary.
```

Reviewer: Instead of `therefore it is not showing in the data summary`, we should include why full-text fields do not appear.

> * the field is a Geo field, therefore it is not showing in the data summary prior to v.7.14.0

Reviewer: It is best practice to avoid calling out versions in the docs because it is hard to maintain. If there are version-specific docs, we only push PRs to those specific versions.

Reviewer: This section is about why fields do not appear in the *Available fields* list, so I'm not sure we need to include information about data summaries.

Reviewer: Is `Geo field` correct? I'm not familiar with Maps, but is this the terminology they use?

Author: `geo_point` field is the naming in the app:

<img width="549" alt="Screenshot 2021-07-14 at 14 43 39" src="https://user-images.githubusercontent.com/924948/125623982-5c884c3c-85e9-43ed-bed4-ac232821ffe3.png">

Author: > This section is about why fields do not appear in the _Available fields_ list, so I'm not sure we need to include information about data summaries.

You are right. Prior to 7.14 is wasn't showing up in the `Available fields`, not data summaries

> * the mapping is not supported by index-patterns, for instance for fields in `flattened` objects

Reviewer: Data analysts are unfamiliar with the concept of mapping. This statement is more for admins. 

I would reword this statement so that it applies to analysts, or remove it.

> If your field does not fall into one of the cases above, check if that is available into the `Empty fields` group: *Lens* uses an heuristic to determine whether the fields have available values or not and for sparse datasets it can be less precise.

Reviewer: ```suggestion
Verify if the field appears in the *Empty fields* list. *Lens* uses heuristics to determine if the fields contain values. For sparse data sets, the heuristics are less precise.
```

** If a custom `Number` configuration is used, check that the color stop values are covering the current data range.

** If a `Percent` configuration is used, and the data range changes, the colors displayed are affected.

[float]
[[is-it-possible-to-show-icons-in-datatable]]

> ===== Is it possible to show icons in datatable?

Reviewer: ```suggestion
===== Is it possible to display icons in data tables?
```

> It is possible to show icons <<managing-index-patterns, via field formatter>>.

Reviewer: ```suggestion
You can display icons with <<managing-index-patterns, field formatter>> in data tables.
```

Reviewer: Can you add an example for when users would want to display an icon in a data table?
